composeml/label_times/object.py

```diff
@@ -17,65 +17,84 @@ def __init__(
         self,
         data=None,
         target_entity=None,
-        name=None,
-        label_type=None,
+        target_types=None,
+        target_columns=None,
         search_settings=None,
         transforms=None,
         *args,
         **kwargs,
     ):
         super().__init__(data=data, *args, **kwargs)
         self.target_entity = target_entity
-        self.label_name = name
-        self.label_type = label_type
+        self.target_columns = target_columns or []
+        self.target_types = target_types or {}
         self.search_settings = search_settings or {}
         self.transforms = transforms or []
         self.plot = LabelPlots(self)
 
-        if not self.empty:
-            self._check_label_name()
-            self._check_label_type()
+        if not self.empty: self._check_label_times()
+
+    def _assert_single_target(self):
+        info = 'must first select an individual target'
+        assert self._is_single_target, info
 
-    def _check_label_name(self):
+    def _check_label_times(self):
         """Checks whether the target exists in the data frame."""
-        if self.label_name is None:
-            self.label_name = self._infer_label_name()
+        if not self.target_columns:
+            self.target_columns = self._infer_target_columns()
+
+        missing = pd.Index(self.target_columns).difference(self.columns)
+        info = 'target variable(s) not found: %s'
+        assert missing.empty, info % missing.tolist()
```

Comment:
I think this code can be place inside an `else` clause of `if not self.target_columns`. Since `_infer_target_columns` is using `self.columns` as a starting point it should never return a target column that can't be found in the original columns.

Reply:
Refactored code to an `else` clause.